buku

```diff
@@ -468,8 +468,39 @@ class BukuDb:
 
         self.conn.commit()
 
+    def delete_tag_at_index(self, index, tag_manual):
+        """ Delete tags for bookmark at index
+
+        :param index: int position of record, 0 for all
+        :tag_manual: string of comma-separated tags to delete manually
+        """
+
+        if index == 0:
+            resp = input('Tags will be deleted for ALL bookmarks. Enter \x1b[1my\x1b[21m to confirm: ')
+            if resp != 'y':
+                return
+
+            self.cur.execute('SELECT id, tags FROM bookmarks ORDER BY id ASC')
```

Comment:
For index == 0, instead of fetching all records, can you please fetch the records matching the tag? Please refer to `search_by_tag` method. This will redundant eliminate redundant error checks.